airflow/providers/common/sql/operators/sql.py

```diff
@@ -221,7 +221,7 @@ def __init__(
         sql: str | list[str],
         autocommit: bool = False,
         parameters: Mapping | Iterable | None = None,
-        handler: Callable[[Any], Any] = fetch_all_handler,
+        handler: Callable[[Any], Iterable[Tuple]] = fetch_all_handler,
```

Comment:
isn't it should be `handler: Callable[[Any], Optional[List[Tuple]]]`?

Reply:
Why optional? If there's no results it will return an empty list. I agree that it should be a list, will change the Iterable to List

Reply:
Fixed